src/main/typescript/components/Header.tsx

```diff
@@ -83,43 +85,58 @@ interface HeaderProps {
     isLoggedIn: boolean
     loginName: string
     signout: () => Action
+    getUser: () => ReduxAction<Promise<UserDetails>>
 }
 
-const Header = ({ isLoggedIn, loginName, signout }: HeaderProps) => {
-    const loginNavBar = isLoggedIn
-        ? [
-            <span key="loginName" className="navbar-text">{loginName}</span>,
-            <NavBarLink key="my datasets" to={depositOverviewRoute}>My Datasets</NavBarLink>,
-            <Link onClick={signout} className="nav-link logoff" key="log out" to={homeRoute} title="Log out">Log out</Link>,
-        ]
-        : [<NavBarLink key="login" to={loginRoute} title="Login to EASY">Login</NavBarLink>]
+class Header extends Component<HeaderProps> {
 
-    return <>
-        <NavBar>
-            <NavBarLink to={homeRoute} title="Home">Home</NavBarLink>
-            <NavBarLink to={registerRoute} title="Register to get access to EASY">Register</NavBarLink>
-            {...loginNavBar}
-        </NavBar>
+    componentDidUpdate() {
+        const { isLoggedIn, loginName, getUser } = this.props
+        if (loginName == undefined) {
```

Comment:
`if (isLoggedIn && !loginName)`

Reply:
note, `loginName === undefined` == `!loginName`

beside, always use `===` instead of `==` in Java/TypeScript